snapshot_manager/snapshot_manager/file_access.py

```diff
@@ -5,19 +5,11 @@
 import logging
 import pathlib
 import tempfile
-import typing
-
-
-@typing.overload
-def write_to_temp_file(content: bytes, **kw_args) -> pathlib.Path: ...
-
-
-@typing.overload
-def write_to_temp_file(text: str, **kw_args) -> pathlib.Path: ...
 
 
 def write_to_temp_file(
-    content: str | bytes, prefix: str = "snapshot-builder-"
+    content: str | bytes,
+    prefix: str | None = "snapshot-builder-",
```

Comment:
With only one parameter it makes more sense to have this `typeA | typeB` annotation.